tools/deployment/k8s/s3/deploy.sh

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env bash
+set -eo pipefail
+
+# AWS CLI Container Image Script
+# Ensure AWS CLI environment setup:
+# - Use official AWS CLI container image (amazon/aws-cli) and ensure curl, tar, gzip, and jq is preinstalled.
+# - Configure AWS credentials using environment variables through various options:
+#   - Put credential files under $HOME/.aws/credentials
+#   - Set AWS_CONFIG_FILE for custom configuration file path.
+#   - Set AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY directly.
+#   - Refer to AWS CLI docs for more authentication options.
+
+# Expected Environment Variables:
+# - AWS_ENDPOINT_URL: Example - "http://minio:9091"
+# - LOG_VIEWER_BUCKET: Example - "log-viewer"
+# - TAG_NAME: Example - "latest" (latest stable release), A GitHub release tag, or <empty> (latest prerelease)
+
+LOG() {
+    echo "$(date -u +"%Y-%m-%dT%H:%M:%SZ") $1" >&2
+}
+
+wait_for_s3_availability() {
+    local max_retries=10
+    local retry_delay=6
+
+    local retries=0
+    while [ $retries -lt $max_retries ]; do
+        if ! aws s3 ls --endpoint-url "$AWS_ENDPOINT_URL" > /dev/null 2>&1; then
+            LOG "[INFO] S3 API endpoint did not return successfully. Retrying in $retry_delay seconds ..."
+            retries=$((retries+1))
+            sleep $retry_delay
+        else
+            break
+        fi
```

Comment:
How about rewriting this to make the success case `break`? Then you don't need to encapsulate the failure case in an `else` block.